src/dotnet/Chat.UI.Blazor/Services/ChatUI.StateSync.cs

```diff
@@ -145,29 +145,45 @@ private async Task SynchronizeSelectedChatIds()
         }
     }
 
-    private async Task RestoreSelectedPlace(CancellationToken cancellationToken)
+    private async Task RestoreSelectedNavbarGroup(CancellationToken cancellationToken)
     {
         try {
             var selectedChatId = await SelectedChatId.Use(cancellationToken).ConfigureAwait(false);
             if (selectedChatId.IsNone)
                 return;
 
+            var chat = await Chats.Get(Session, selectedChatId, cancellationToken).ConfigureAwait(false);
+            if (chat == null)
+                return;
+
+            var isChatsSelected = NavbarUI.IsGroupSelected(NavbarGroupIds.Chats);
```

Comment:
IMHO, it makes sense to extract common part for this piece and almost the same code below